src/main/java/frc/robot/subsystems/hang/docs/Hang.md

```diff
@@ -0,0 +1,51 @@
+# Hang subsystem
+## description
+This subsystem allows the frc robot to hang on the chain in the stage.
+
+
+##connected devices
+a motor (can sparkmax)
+
+
+## Methods
+| Method Name | Returns | Parameters | Description | Notes |
+| --- | --- | ---- | ---- | --- |
+|getInstance|instance of hang |N/A|creates a new version of the hang|    |
+|hang|void|N/A|once instance is made, creates a constructor for hang|   |
+|setupMotors|void|N/A|starts initial code.| |
+|setPID|void|p,i,d|determines how "smoothly" the motor for the hang, runs.| |
+|periodic|void|N/A|runs codes repeatedly within the method| |
+|doSendables|void|N/A|logs data from hang| |
+|setManualVelocity|void|velocity|sets the velocity of the motor that we decide| |
+|getHangPosition|double|N/A|finds the positon of the hang motor|angle|
+|setTargetPosition|void|targetPosition|sets the postion we that want the motor to get to/be at| |
+|isAtPosition|void|N/A|determines if the motor is at the needed poistion| |
+|getError|double|N/A|determines how far off the motor is from where it should be|current position-target position|
+|getHangVelocity|double|N/A|finds the velocity of the motor| |
+|getMotorCurrent|double|N/A|find the currunent being outputed in the motor|in amps|
+|setHangMode|void|HangControlType type|figures out the hang mode to determine what is happening| |
```

Comment:
look up enums. We have a HangControlType enum that determines whether the hang should run at a constant voltage/speed (Manual mode) or run directly to a position using PID (PID mode).